File: Data Structures & Algorithms/two-integer-sum-ii/submission-13.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        var l = 0;
        var r = numbers.Length - 1;
        while (l < r) {
            var current = numbers[l] + numbers[r];
            if (current < target) {
                l++;
            } 
            else if (current > target) 
            {
                r--;
            }
            else {
                return [l+1, r + 1];
            }
        }
        return [];
    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-13.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        store = set()
        current = head
        while current and current.next:
            if current in store:
                return True
            store.add(current)
            current = current.next
        return False